cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/api/application"
+	"github.com/juju/juju/api/charms"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/juju/block"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// NewUpdateSeriesCommand returns a command which updates the series of
+// an application or machine.
+func NewUpdateSeriesCommand() cmd.Command {
+	return modelcmd.Wrap(&updateSeriesCommand{})
+}
+
+// updateApplicationAPI defines a subset of the application facade, as required
+// by the update-series command.
+type updateSeriesAPI interface {
+	Get(string) (*params.ApplicationGetResults, error)
+	GetCharmURL(string) (*charm.URL, error)
+	UpdateApplicationSeries(string, string, bool) error
+}
+
+// charmAPI defines a subset of the charm facade, as required
+// by the update-series command.
+type charmAPI interface {
+	CharmInfo(charmURL string) (*charms.CharmInfo, error)
+}
+
+// updateSeriesCommand is responsible for updating the series of an application or machine.
+type updateSeriesCommand struct {
+	modelcmd.ModelCommandBase
+
+	apiRoot            api.Connection
+	charmClient        charmAPI
+	updateSeriesClient updateSeriesAPI
+
+	applicationName string
+	force           bool
+	machineNumber   string
+	series          string
+}
+
+var updateSeriesDoc = `
+It is recommended to update the series on existing units before running this
```

Comment:
not sure about this - maybe leave it out for now
also, first sentence(s) should explain what the command does, then follow with any caveats etc